src/inmanta/deploy/scheduler.py

```diff
@@ -357,6 +382,43 @@ async def _run_for_agent(self, agent: str) -> None:
 
             self._work.agent_queues.task_done(agent, task)
 
+    async def is_running(self) -> bool:
+        current_environment = await Environment.get_by_id(self.environment)
+        assert current_environment
+        requested_agent = await data.Agent.get(env=self.environment, endpoint=const.AGENT_SCHEDULER_ID)
+        return not (current_environment.halted or requested_agent.paused)
+
+    async def refresh_agent_state_from_db(self, name: str) -> None:
+        """
+        Refresh from the DB (authoritative entity) the actual state of the agent.
+            - If the agent is not paused: It will make sure that the agent is running.
+            - If the agent is paused: Stop a particular agent.
+
+        :param name: The name of the agent
+        """
+        requested_agent = await data.Agent.get(env=self.environment, endpoint=name)
+
+        # This should never occur as the Scheduler is creating these 'agents'!
+        if requested_agent and name not in self._state.agent_status:
+            raise RuntimeError("Agent exists in Database but not in the Scheduler state!")
+
+        if name not in self._state.agent_status and requested_agent is None:
+            raise LookupError(f"The agent `{name}` does not exist!")
```

Comment:
How do you know? You have no control over when this method is called? (The server calls it)